### PR TITLE
Push docker images to DockerHub

### PR DESCRIPTION
**Pull request by jackpoz**
**Changes proposed:**

-  Add new Circle CI Environmental Variable (not GitHub) DOCKERHUB_PUSH_IMAGES to enable/disable pushing Docker images created by Circle CI to DockerHub, disabled by default. Requires "TRUE" as value to be enabled.
- Filter branches pushed to "3.3.5" and "master"
- Tag images with branch name and commit hash
- Each fork will push to its own dockerhub repository (if enabled)
- Change Circle CI pch build from Debug -DNDEBUG to Release (RelWithDebInfo slowed the build by 10 more minutes than just Release and pushing Debug images means none will use them) when DOCKERHUB_PUSH_IMAGES is enabled. Existing forks that will not set DOCKERHUB_PUSH_IMAGES will not be affected by this change.
**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:**

None

**Tests performed:**

- Ran Circle CI a few times, you can see the pushed images at https://hub.docker.com/r/jackpoz/trinitycore/tags?page=1&ordering=last_updated
- Did a pull locally
image

**Known issues and TODO list:** (add/remove lines as needed)

None

### CODE REVIEW

**Comment by jackpoz** [comment]
It could be a better idea to push nightly builds instead for 3.3.5 and master

**Comment by chipzz** [comment]
You could also distinguish between actual commits (build RelWithDebInfo) and test builds for PRs (build Release, or even  Debug -DNDEBUG)?

**Comment by jackpoz** [comment]
@illfated I updated the readme for docker, could you please take a look as see if there are some adjustments about grammar and syntax and whatnot ? thanks 😃

File: contrib/Docker/README.md [hunk 1]
```diff
@@ -1,8 +1,25 @@
 # Docker
 
-The Circle CI Linux pch job uses the Dockerfile contained in the same folder as this README to create an image with the binaries built for Linux, and stores that in the job artifacts.
+The Circle CI Linux pch job uses the Dockerfile contained in the same folder as this README to create an image with the binaries built for Linux, and stores that in the job artifacts. For 3.3.5 and master branches, it also pushes the images to https://hub.docker.com/r/trinitycore/trinitycore
```
Comment (illfated) [inline contrib/Docker/README.md:3]:
```suggestion
The Circle CI Linux pch job uses the Dockerfile contained in the same folder as this README to create an image with the binaries built for Linux, and stores that in the job artifacts. For the 3.3.5 and master branches, it also pushes the images to https://hub.docker.com/r/trinitycore/trinitycore .
```

File: contrib/Docker/README.md [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
 
-The instructions below expect a basic knowledge of how to configure TrinityCore and how to use Docker:
+The instructions below expect a basic knowledge of how to configure TrinityCore and how to use Docker.
+
+## Load the docker image
+For 3.3.5 and master branches, it's possible to pull the images from DockerHub.
+- For latest 3.3.5 use the following command:
```
Comment (illfated) [inline contrib/Docker/README.md:9]:
```suggestion
## Load the Docker image
For the 3.3.5 and master branches, it's possible to pull the images from DockerHub.
- For latest 3.3.5, use the following command:
```

File: contrib/Docker/README.md [hunk 3]
```diff
@@ -1,8 +1,25 @@
...
+  ```
+  docker pull trinitycore/trinitycore:3.3.5
+  ```
+- For latest master use the following command:
```
Comment (illfated) [inline contrib/Docker/README.md:13]:
```suggestion
- For latest master, use the following command:
```

File: contrib/Docker/README.md [hunk 4]
```diff
@@ -1,8 +1,25 @@
...
+  ```
+  docker pull trinitycore/trinitycore:master
+  ```
+- For a specific 3.3.5 or master commit, use the following command, replacing "commit_hash" with the hash of the commit:
+  ```
+  docker pull trinitycore/trinitycore:commit_hash
+  ```
+
+For Pull Requests or branches other than 3.3.5 or master, follow the steps below to load the image from Circle CI
```
Comment (illfated) [inline contrib/Docker/README.md:22]:
```suggestion
For Pull Requests or branches other than 3.3.5 or master, follow the steps below to load the image from Circle CI:
```

File: contrib/Docker/README.md [hunk 5]
```diff
@@ -13,6 +30,7 @@ The instructions below expect a basic knowledge of how to configure TrinityCore
     docker load -i docker.tar.gz
     ```
 
+## Start authserver/worldserver from docker
```
Comment (illfated) [inline contrib/Docker/README.md:33]:
```suggestion
## Start authserver/worldserver from Docker
```

File: contrib/Docker/README.md [hunk 6]
```diff
@@ -13,6 +30,7 @@ The instructions below expect a basic knowledge of how to configure TrinityCore
...
 1. Copy the .conf files from the TrinityCore GitHub repository to a local folder which will be passed on as a mapped volume to docker.
```
Comment (illfated) [inline contrib/Docker/README.md:34]:
```suggestion
1. Copy the .conf files from the TrinityCore GitHub repository to a local folder which will be passed on as a mapped volume to Docker.
```

**Comment by illfated** [comment]
Ref. https://www.docker.com/legal/trademark-guidelines:

### Use of capitals
Our Marks should be written with the proper capitalization as shown in these guidelines. For example: “Docker”. Docker may be written in lower case as ‘docker’ only when the use explicitly refers to command line commands. For example in documentation, help text or examples of use.

**Comment by illfated** [comment]
2 lines out of reach for comments / suggestions:

Maybe cosmetic, but I think the initialism "SQL" should be capitalized, because it is a common acronym:
```diff
- 1. Download the TDB sql file from GitHub.
+ 1. Download the TDB SQL file from GitHub.
```

Last occurrence of "docker" I could find, maybe you view it as a technical phrase here (I am uncertain):
```diff
- 1. Download the docker.tar.gz archive containing the docker image.
+ 1. Download the docker.tar.gz archive containing the Docker image.
```

**Comment by jackpoz** [comment]
Well for "sql file" I mean the file extension, not the language

**Comment by illfated** [comment]
OK, fair enough. Very well.

**Review by ghost — APPROVED** [review]
Looks good to me, based on the few areas of expertise in which I am able to contribute.

**Pull request merged**